src/Middleware/ResponseCaching/src/ResponseCachingOptions.cs

```diff
@@ -30,4 +30,9 @@ public class ResponseCachingOptions
     /// For testing purposes only.
     /// </summary>
     internal TimeProvider TimeProvider { get; set; } = TimeProvider.System;
+
+    /// <summary>
+    /// <c>true</c> if caching is allowed for authorised endpoints
+    /// </summary>
+    public bool AllowAuthorizedEndpoint { get; set; } = false;
```

Comment:
Changes that add new public APIs need an approved API Proposal: https://github.com/dotnet/aspnetcore/issues/new/choose

Authorised requests are typically not cached for security reasons.

If this was an accepted proposal you'd almost certainly need to implement tests.